docs/02-guides/server-side-events/create-an-event.mdx

```diff
@@ -1,15 +1,158 @@
+
+```tsx title="app/server-events/events/QuestionVotedEvent.ts"
+import { ServerEvent } from "@front-commerce/core/events";
+
+// We define a static event Name for our event
+const QuestionVotedEventName = "QuestionVotedEvent";
+
+// Definition of the payload
+type QuestionVotedEventPayload = {
+  questionSlug: string;
+  upVote: boolean;
+};
+
+// We define an event class that extends the ServerEvent class
+// This class will handle data to be pushed to the server event pipeline
+export default class QuestionVotedEvent extends ServerEvent<QuestionVotedEventPayload> {
+  static event_type = QuestionVotedEventName;
+
+  // We define the constructor for our event
+  // It will call the super constructor (constructor of ServerEvent)
+  // to dispatch event to the server event pipeline
+  constructor(questionSlug: string, upVote: boolean, shopId: string) {
+    super(
+      QuestionVotedEventName,
+      new Date(),
+      // This are data that we want to gather
+      // The question
+      // If the question has been upvoted or downvoted
+      { questionSlug, upVote },
+      // This are the event metadata, they are relative to the event
+      // but don't contain event data, here we collect the shopId
+      // to determine on which shop the question has been voted
+      { shopId }
+    );
+  }
+}
+```
+
+:::info
+
+You can checkout the
+[ServerEvent Reference](/docs/3.x/api-reference/front-commerce-core/ServerEvent)
+to learn more about the usage of ServerEvent class.
 
 :::
+
+That's it!
+
+## Use the Event
+
+Now that we've created a Domain event class, let's use it in a route action:
+
+```tsx title="app/routes/_main.faq.$slug.tsx"
+...
+export const action = async ({
+  context,
+  params,
+  request,
+}: ActionFunctionArgs) => {
+  const { slug } = params;
+
+  const formData = await request.formData();
+  // The `action` uses the `FrontCommerceApp` class to access the Front-Commerce
+  // core from Remix. In this example, we use it to mutate data in the GraphQL
+  // unified API (similar to the one you're used to).
+  const app = new FrontCommerceApp(context.frontCommerce);
+  const response = await app.graphql.mutate(VoteIsFaqUsefulDocument, {
+    input: {
+      slug,
+      isFaqUseful: formData.get("isFaqUseful") === "true",
+    },
+  });
+
+  // highlight-start
+  // We call the `ServerEventBus` to dispatch an event.
+  // In this example, we use it to dispatch a `QuestionVotedEvent` event.
+  // We will instanciate a `QuestionVotedEvent` class with the data we want to
+  // collect for this event.
+  app.services.ServerEventBus.publish(
+    new QuestionVotedEvent(
+      String(slug),
+      formData.get("isFaqUseful") === "true",
+      app.config.currentShopId
+    )
+  );
+  // highlight-end
+
+  if (!response.voteIsFaqUseful) {
+    throw new Response("Question not found", { status: 404 });
+  }
+
+  return json({
+    success: response.voteIsFaqUseful?.success ?? false,
+  });
+};
+...
+```
```

Comment:
**issue:** I don't think this highlight makes sense.

In the prerequisites section, you mention that the reader can directly use the `FAQ extension` example if they want to see the server events in action **without any code changes**. This implies that everything in this guide is **new** to their project, as in, they'll have to create the entire `app/routes/_main.faq.$slug.tsx` file for this to work.

**suggestion:** remove all mention of code coming from the FAQ extension example, and focus on what's actually necessary for the event to be used correctly. The following suggestion is _just_ a suggestion, and should be reworked as you see fit.

```suggestion
```tsx title="app/routes/_main.faq.$slug.tsx"
// ...
export const action = async ({
  context,
  params,
  request,
}: ActionFunctionArgs) => {
  const { slug } = params;
  const app = new FrontCommerceApp(context.frontCommerce);

  // ... Here should be the "voting" mechanism (probably a mutation)
  
  // We call the `ServerEventBus` to dispatch an event.
  // In this example, we use it to dispatch a `QuestionVotedEvent` event created above.
  app.services.ServerEventBus.publish(
    new QuestionVotedEvent(
      String(slug),
      formData.get("isFaqUseful") === "true",
      app.config.currentShopId
    )
  );

  // ... The rest of the action, return, etc.
};
...
```
```

Reply:
fixed with your suggestions